db/migration/migration.go

```diff
@@ -78,30 +79,67 @@ func Init() error {
 
 	logger.Info("migration path: %q, version: %d", migrationsPath, v)
 
-	return nil
+	if err = goose.LoadMigrationPlugins(migrationsPath); err != nil {
+		return fmt.Errorf("migration: failed to load migration plugins: %s", err)
+	}
+
+	if noInit {
+		// pending migrations are not allowed if no-init is enabled (no_init=true)
+		last, err := goose.LastMigration(migrationsPath)
+
+		if err != nil {
+			if err != goose.ErrNoNextVersion {
+				return fmt.Errorf("migration: %s", err)
```

Comment:
1. put some (or no) files in the migration directory
2. mock goose.GetDBVersion and goose.Status(db, migrationsPath)
3. run migration.Init and make asserts on the result